iceoryx_hoofs/include/iceoryx_hoofs/cxx/scope_guard.hpp

```diff
@@ -43,7 +43,7 @@ namespace cxx
 /// }
 /// @endcode
 template <uint64_t CleanupCapacity = DEFAULT_FUNCTION_CAPACITY>
-class ScopeGuardWithVariableCapacity
+class ScopeGuardWithVariableCapacity final
```

Comment:
Why do we need the final specifier here? In theory it could be useful to extend it and I do not see it as strictly needed here. 

We do not have it for most other classes as well and I would argue it is not a good style to do so unless there is a reason. One reason is compiler optimization when a virtual base virtual is involved, but this does not apply here ...

Reply:
The class includes a public d'tor that is not virtual and therefore should not be used as a base class. Once we need the class as a base class, we can remove the final keyword and make the d'tor virtual.

Reply:
Yes, but this is only an issue if a potential derived class is used polymorphically (i.e. trying to invoke the dtor by a reference or pointer to base). Otherwise it is perfectly safe to derive, but of course not possible once the potential base class is declared `final`.

I am fine with this, but I find it unnecessary if classes are designed correctly. (STL classes are also not final nor do they have a virtual dtor I think). virtual dtors also come  with a cost (virtual table).